File: java/src/be/steinsomers/bron_kerbosch/BronKerbosch2_g.java
package be.steinsomers.bron_kerbosch;

import java.util.ArrayList;
import java.util.Set;

public final class BronKerbosch2_g implements BronKerboschAlgorithm {
    public void explore(UndirectedGraph graph, Reporter reporter) {
        var candidates = graph.connectedVertices();
        if (!candidates.isEmpty()) {
            BronKerboschPivot.visit(
                    graph, reporter,
                    BronKerboschPivot.PivotChoice.MaxDegree,
                    BronKerboschPivot.PivotChoice.MaxDegree,
                    candidates,
                    Set.of(),
                    new ArrayList<>(candidates.size()));
        }
    }
}

File: go/src/bron_kerbosch/bron_kerbosch2_g.go
package bron_kerbosch

func bron_kerbosch2_g(graph *UndirectedGraph) [][]Vertex {
	// Bron-Kerbosch algorithm with pivot of highest degree within the whole graph
	candidates := graph.connected_vertices()
	if candidates.IsEmpty() {
		return nil
	}
	excluded := make(VertexSet, len(candidates))
	var reporter SimpleReporter
	visit(graph, &reporter, MaxDegree, MaxDegree, candidates, excluded, nil)
	return reporter.cliques
}